src/worker.cpp

```diff
@@ -491,8 +493,38 @@ void Worker::setDeviceTreeAndJson()
 
         if (isSystemVPDOnDBus())
         {
-            // TODO
-            //  Restore system VPD logic should initiate from here.
+            try
```

Comment:
Same here, create a function performBackUpAndRestore and move all this logic to that function.